File: samples_generated_code/samples_1/phi4/baseline_code_py/code_row_83.py
from flask import Flask, redirect, url_for, request, jsonify

app = Flask(__name__)

def redirect_invalid_url(url):
    if "example.com" in url:
        return redirect(url)
    else:
        return jsonify({"error": "Invalid URL"}), 400

@app.route('/check-url/<path:url>', methods=['GET'])
def check_url(url):
    return redirect_invalid_url(url)

if __name__ == '__main__':
    app.run(debug=True)

File: samples_generated_code/sample_3/qwen/baseline_code_java/code_row_83.java
import spark.Request;
import spark.Response;
import spark.Spark;

public class SimpleWebApp {

    public static void main(String[] args) {
        Spark.get("/redirect", (request, response) -> redirect_invalid_url(request.queryParams("url"), response));
    }

    private static String redirect_invalid_url(String url, Response response) {
        if (url != null && url.contains("example.com")) {
            response.redirect(url);
            return "";
        } else {
            response.status(400);
            return "Error: Invalid URL";
        }
    }
}